src/mame/misc/magic10.cpp

```diff
@@ -737,6 +825,62 @@ static INPUT_PORTS_START( sgsafari )
 	PORT_DIPSETTING(        0x0000, DEF_STR( On ) )
 INPUT_PORTS_END
 
+static INPUT_PORTS_START( spetrix )
+	PORT_START("IN0")
+	PORT_BIT( 0x0001, IP_ACTIVE_LOW, IPT_POKER_HOLD1 )   PORT_NAME("Right - Hold 1")
+	PORT_BIT( 0x0002, IP_ACTIVE_LOW, IPT_POKER_HOLD2 )   PORT_NAME("Left - Hold 2 / Low")
+	PORT_BIT( 0x0004, IP_ACTIVE_LOW, IPT_POKER_HOLD3 )   PORT_NAME("Up - Hold 3")
+	PORT_BIT( 0x0008, IP_ACTIVE_LOW, IPT_POKER_HOLD4 )   PORT_NAME("Down - Hold 4 / High")
+	PORT_BIT( 0x0010, IP_ACTIVE_LOW, IPT_POKER_HOLD5 )   PORT_NAME("Rotate Left - Hold 5")
+	PORT_BIT( 0x0020, IP_ACTIVE_LOW, IPT_START1 )        PORT_NAME("Select - Start / Collect")
+	PORT_BIT( 0x0040, IP_ACTIVE_LOW, IPT_GAMBLE_BET )    PORT_NAME("Rotate Right - Bet / Collect")
+	PORT_BIT( 0x0080, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x0100, IP_ACTIVE_LOW, IPT_COIN1 )         PORT_NAME("Coin 1")
+	PORT_BIT( 0x0200, IP_ACTIVE_LOW, IPT_COIN2 )         PORT_NAME("Coin 2")
+	PORT_BIT( 0x0400, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x0800, IP_ACTIVE_LOW, IPT_START2 )        PORT_NAME("Petrix Start")
+	PORT_SERVICE_NO_TOGGLE( 0x1000, IP_ACTIVE_LOW )
+	PORT_BIT( 0x2000, IP_ACTIVE_LOW, IPT_CUSTOM )        PORT_READ_LINE_DEVICE_MEMBER("ticket", ticket_dispenser_device, line_r)
+	PORT_BIT( 0x4000, IP_ACTIVE_LOW, IPT_UNUSED )
+	PORT_BIT( 0x8000, IP_ACTIVE_LOW, IPT_GAMBLE_PAYOUT ) PORT_NAME("Collect Points")
+
+	PORT_START("DSW1")
+	PORT_BIT( 0x0001, IP_ACTIVE_HIGH, IPT_UNUSED )
+	PORT_BIT( 0x0002, IP_ACTIVE_HIGH, IPT_UNUSED )
+	PORT_BIT( 0x0004, IP_ACTIVE_HIGH, IPT_UNUSED )
+	PORT_BIT( 0x0008, IP_ACTIVE_HIGH, IPT_UNUSED )
+	PORT_BIT( 0x0010, IP_ACTIVE_LOW,  IPT_BUTTON1 )      PORT_NAME("Boot to Poker")
```

Comment:
Does pressing this switch mode to poker, or does it have to be set at power-on to boot to poker?

For the former, I’d suggest `IPT_SERVICE1` would be more appropriate.

For the latter, it looks more like a candidate for `PORT_CONFNAME` etc. or `IPT_OTHER` along with `PORT_TOGGLE`.

Reply:
The switch is only for use by the operator/installer.
When pressed the machine locks with a black screen until it’s released. When released the machine runs the poker game. From now on the poker game is permanently selected even if the machine is reset or turned off/on. There is no way of going back to Tetris.
It can be pressed at any time.

Reply:
If it’s an operator control, `IPT_SERVICE1` is a better choice.  A better name might be “Switch to Poker Mode” or similar.